### PR TITLE
Ingress space-cli deleters

### LINKED ISSUE #1155: Add a space-cli delete <resource> command
### The problem faced currently?
Currently we can apply config to SpaceCloud. This solves the need to create and update resources. However, there is no mechanism to delete a resource via. space-cli.
### How can we solve it?
We can add `deleters` for each resource. This will be similar to the `generators` and `getters`. We will have to be careful to make sure we give flexibility to delete resources by `id`, `prefix` (this can be a multi select) or delete all resources.
>If you want this feature to be implemented, give it a thumbs up reaction, so that we can determine which features are important to you.
>👍

### PR DESCRIPTION
**Pull request by KavishShah09**
fixes #1155 for ingress module

### CODE REVIEW

File: gateway/managers/syncman/syncman_routing.go [hunk 1]
```diff
@@ -174,6 +174,33 @@ func (s *Manager) SetGlobalRouteConfig(ctx context.Context, project string, glob
...
+	s.modules.Routing().SetGlobalConfig(projectConfig.IngressGlobal)
+
+	// Finally lets store the config
+	resourceID := config.GenerateResourceID(s.clusterID, project, config.ResourceIngressGlobal, "global")
+	if err := s.store.DeleteResource(ctx, resourceID); err != nil {
+		return http.StatusInternalServerError, err
+	}
+
+	return http.StatusOK, nil
+}
```
Comment (sharadregoti) [inline gateway/managers/syncman/syncman_routing.go:202]: Not required, use setter endpoints to set empty values

**Review by sharadregoti — CHANGES_REQUESTED** [review]

File: space-cli/cmd/modules/ingress/commands.go [hunk 2]
```diff
@@ -126,3 +111,48 @@ func actionGetIngressGlobal(cmd *cobra.Command, args []string) error {
...
+
+// DeleteSubCommands is the list of commands the ingress module exposes
+func DeleteSubCommands() []*cobra.Command {
+
+	var deleteRoutes = &cobra.Command{
+		Use:               "ingress-routes",
+		Aliases:           []string{"ingress-route"},
+		RunE:              actionDeleteIngressRoutes,
+		ValidArgsFunction: ingressRoutesAutoCompleteFun,
+		Example:           "space-cli delete ingress-routes routeID --project myproject --log-level info",
```
Comment (sharadregoti) [inline space-cli/cmd/modules/ingress/commands.go:123]: remove --log-level info, from all commands

File: space-cli/cmd/modules/ingress/delete.go [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+func deleteIngressRoute(project, prefix string) error {
+
+	objs, err := GetIngressRoutes(project, "ingress-route", map[string]string{}, []string{})
+	if err != nil {
+		return err
+	}
+
+	doesRouteExist := false
```
Comment (sharadregoti) [inline space-cli/cmd/modules/ingress/delete.go:32]: What is the use of this boolean?

File: space-cli/cmd/modules/ingress/delete.go [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+	routeIDs := []string{}
+	for _, spec := range objs {
+		routeIDs = append(routeIDs, spec.Meta["id"])
+	}
+
+	prefix, err = filter.DeleteOptions(prefix, routeIDs, doesRouteExist)
```
Comment (sharadregoti) [inline space-cli/cmd/modules/ingress/delete.go:38]: prefix sounds misleading, call it resourceID
Do this every where

File: space-cli/cmd/modules/ingress/delete.go [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+	"github.com/spaceuptech/space-cloud/space-cli/cmd/utils/transport"
+)
+
+func deleteIngressGlobalConfig(project string) error {
+
+	// Delete the ingress global config from the server
+	url := fmt.Sprintf("/v1/config/projects/%s/routing/ingress/global", project)
+
+	payload := new(model.Response)
+	if err := transport.Client.MakeHTTPRequest(http.MethodPost, url, map[string]string{}, payload); err != nil {
```
Comment (sharadregoti) [inline space-cli/cmd/modules/ingress/delete.go:18]:
```suggestion
	if err := transport.Client.MakeHTTPRequest(http.MethodPost, url, map[string]string{}, new(model.Response)); err != nil {
```
Do this everywhere, where you are not using the payload

**Review by sharadregoti — CHANGES_REQUESTED** [review]

**Comment by sharadregoti** [comment]
Test this once, you don't need Kubernetes for this

File: space-cli/cmd/modules/ingress/delete.go (hunk 3 above)
Comment (KavishShah09) [reply space-cli/cmd/modules/ingress/delete.go:32]: Used in differentiating when to survey only filtered resource IDs and when to show all

Comment (sharadregoti) [reply space-cli/cmd/modules/ingress/delete.go:31]: You don't need to pass this as a parameter to the function. Just create the boolean locally inside the Deleteoptions function.

The boolean that you are passing as a parameter is always false

**Review by sharadregoti — APPROVED** [review]

**Pull request merged**